RDAPI-22611 yamles CLI upgrade documentation (#1837)
Axway/axway-open-docs: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_upgrade.md

```
./yamles upgrade --source yaml:file:///home/user/yaml --output-dir /home/user/yaml-upgraded
```

**Example 3:**

* Specify the `.tar.gz` file for the YAML configuration to upgrade.
* Add the upgraded YAML configuration to the `/home/user/yaml-upgraded` directory.
* Create a `.tar.gz` file of the content of `/home/user/yaml-upgraded`.

```
./yamles upgrade --source yaml-to-upgrade.tar.gz --output-dir /home/user/yaml-upgraded --targz /home/user/archives/yaml-upgraded.tar.gz
```

**Example 4:**

* Specify the `.tar.gz` file for the YAML configuration to upgrade.
* Upgrade the YAML configuration with a non-default passphrase.
* Add the upgraded YAML configuration to the `/home/user/yaml-upgraded` directory.
* Create a `.tar.gz` file of the content of `/home/user/yaml-upgraded`.

```
./yamles upgrade --source yaml-to-upgrade.tar.gz --passphrase secret --output-dir /home/user/yaml-upgraded --targz /home/user/archives/yaml-upgraded.tar.gz
```

{{< alert title="Note">}}
The YAML configuration upgraded is encrypted with the same passphrase as the source YAML configuration.
{{< /alert >}}

**Example 5:**

> * Specify the `.tar.gz` file for the YAML configuration to upgrade.

Reviewer: The 4 sentences on example 5 duplicate the same sentences from example 4. What's going on?

{
"title": "Upgrade YAML configuration using CLI",
"linkTitle": "Upgrade YAML configuration using CLI",
"weight":"80",
"date": "2021-04-15",
"description": "Learn how to use the YAML configuration CLI to upgrade a YAML configuration."
}

> The `upgrade` option may be used to upgrade YAML configurations. Both full configurations, and configuration fragments may be upgraded in the same way.

Reviewer 2: ```suggestion
The `upgrade` option may be used to upgrade YAML configurations. Both full configurations, and configuration fragments can be upgraded the same way.
```

{{% alert color="warning" title="Upgrade Caution" %}}

> * YAML configurations upgrade is supported starting with version 7.7 May 2021.

Reviewer 2: It is actually working for older versions. I think it is Jan21 (the one that containing {{file...}} to replace #ref[binary]). TBD.

Author: Technically speaking yes, it is working since Jan21 version. From a end-user perspective I meant to start the upgrade support since the GA. Are we sure we want to support from Jan21 ?

Author: As discussed, the support mechanism will be available beginning with GA version May21

* If you need to upgrade an old YAML configuration dated before May 2021, you can upgrade the XML federated configuration using [upgradeconfig](/docs/apim_installation/apigw_upgrade/upgrade_analytics/#upgradeconfig-options) or [projupgrade](/docs/apim_reference/devopstools_ref/#projupgrade-command-options) and convert it again to YAML configuration using the latest YAML CLI [fed2yaml](/docs/apim_yamles/apim_yamles_cli/yamles_cli_convert/#convert-your-xml-configuration-to-a-yaml-configuration) or [frag2yaml](/docs/apim_yamles/apim_yamles_cli/yamles_cli_convert/#convert-your-xml-configuration-fragment-to-a-yaml-configuration-fragment).

{{% /alert %}}

## Upgrade a YAML configuration

> The `upgrade` option has the same goal as the [projupgrade](/docs/apim_reference/devopstools_ref/) tool existing for XML federated configurations. It enables you to upgrade YAML configurations or YAML configuration fragments from earlier versions (7.7 May2021 and later) to the latest. The `upgrade` option is the only entry point available to upgrade YAML configuration.

Reviewer 2: ```suggestion
The `upgrade` option has the same goal as the [projupgrade](/docs/apim_reference/devopstools_ref/) tool existing for XML federated configurations. It enables you to upgrade YAML configurations or YAML configuration fragments from earlier versions (7.7 May2021 and later) to the latest. The `upgrade` option is the centralised entry point to upgrade any kind of YAML configuration.
```

**Example 1:**

* Specify the directory for the YAML configuration to upgrade.
* Add the upgraded YAML configuration to the `/home/user/yaml-upgraded` directory.
* Create a `.tar.gz` file of the content of `/home/user/yaml-upgraded`.

```

> ./yamles upgrade --source /home/user/yaml --output-dir /home/user/yaml-upgraded --targz yaml-upgraded.tar.gz

Reviewer 2: Is `--targz` option relative to output-dir ?
* if yes, it is not consistent with other CLI options 
* if no, maybe you should add an absolute path before the .tar.gz

Author: No it is not relative to output-dir. I'll update to put an absolute path.

```

**Example 2:**

* Specify the URL for the YAML configuration to upgrade.
* Add the upgraded YAML configuration to the `/home/user/yaml-upgraded` directory.

```
./yamles upgrade --source yaml:file:///home/user/yaml --output-dir /home/user/yaml-upgraded
```

**Example 3:**

* Specify the `.tar.gz` file for the YAML configuration to upgrade.
* Add the upgraded YAML configuration to the `/home/user/yaml-upgraded` directory.
* Create a `.tar.gz` file of the content of `/home/user/yaml-upgraded`.

```

> ./yamles upgrade --source yaml-to-upgrade.tar.gz --output-dir /home/user/yaml-upgraded --targz yaml-upgraded.tar.gz

Reviewer 2: same remarks here, it suggest targz is not an absolute path

Author: No it is not relative to output-dir. I'll update to put an absolute path.

```

**Example 4:**

* Specify the `.tar.gz` file for the YAML configuration to upgrade.
* Upgrade the YAML configuration with a non-default passphrase.
* Add the upgraded YAML configuration to the `/home/user/yaml-upgraded` directory.
* Create a `.tar.gz` file of the content of `/home/user/yaml-upgraded`.

```

> ./yamles upgrade --source yaml-to-upgrade.tar.gz --passphrase secret --output-dir /home/user/yaml-upgraded --targz yaml-upgraded.tar.gz

Reviewer 2: same as above

Author: No it is not relative to output-dir. I'll update to put an absolute path.
